classes/CategoryHelper.php

```diff
@@ -0,0 +1,97 @@
+<?php
+/**
+ * Contao Open Source CMS
+ *
+ * Copyright (c) 2017 Heimrich & Hannot GmbH
+ *
+ * @author  Rico Kaltofen <r.kaltofen@heimrich-hannot.de>
+ * @license http://www.gnu.org/licences/lgpl-3.0.html LGPL
+ */
+
+namespace NewsCategories;
+
+class CategoryHelper
+{
+
+    /**
+     * Prepare the category object
+     *
+     * @param NewsCategoryModel $objCategory
+     *
+     * @return object
+     */
+    public static function prepareCategory(NewsCategoryModel $objCategory)
+    {
+        $category                = (object) $objCategory->row();
+        $category->name          = $category->frontendTitle ?: $category->title;
+        $category->class         = 'category_' . $category->id . ($category->cssClass ? (' ' . $category->cssClass) : '');
+        $category->linkTitle     = specialchars($category->frontendTitle ?: $category->title);
+        $category->href          = '';
+        $category->hrefWithParam = '';
+        $category->targetPage    = null;
+
+        // Add the target page
+        if (($targetPage = $objCategory->getTargetPage()) !== null)
+        {
+            $category->href          = $targetPage->getFrontendUrl();
+            $category->hrefWithParam = $targetPage->getFrontendUrl('/' . NewsCategories::getParameterName() . '/' . $category->alias);
+            $category->targetPage    = $targetPage;
+        }
+
+        // Register a function to generate category URL manually
+        $category->getUrl = function (\PageModel $page) use ($category) {
+            return $category->getUrl($page);
+        };
+
+        $category->tree = CategoryHelper::getCategoryTree($objCategory->id);
+
+        return $category;
+    }
+
+    /**
+     * Get the category tree with all parent categories of the given category id
+     *
+     * @param integer $intId     The category id
+     * @param bool    $max_level Maximum level of parent categories that should be covered, set to null for unlimited execution, 0 for only current category with its parent
+     * @param array   $all       Required for recursion
+     *
+     * @return array|null
+     */
+    public static function getCategoryTree($intId, $max_level = null, $all = [])
+    {
+        if (!is_numeric($max_level))
```

Comment:
I don't think this condition is necessary if the docblock specifies the type. In the future we will force the argument to be `integer` anyway.